### ISSUE TITLE
[RFC] include a new dev config file for `twig-cs-fixer` in the codebase

### ISSUE THREAD

**Issue by 94noni** [symfony/symfony#54799]
### Description

Hello

May I propose to add the tool `twig-cs-fixer` ([repo](https://github.com/VincentLanglet/Twig-CS-Fixer)) in this repo
Like adding a `twig-cs-fix` config file in the repo as its done for the `php-cs-fixer`
And from time to time, run it manually or via CI

RFC 

What?

- this tool is like the `php-cs-fixer`, but for twig files
- it is based on official twig coding standard, and allows to lint, and also fix, your twig files

Why?

- it is listed on twig official documentation
- it is solid and working great
- it adds value as code standard, as per all other cs fixers

How?

- We can add it, starting from 7.2, make it run on symfony/symfony twig files present in the codebase

Thank you!

---

friendly ping @VincentLanglet and our discussion [here](https://github.com/VincentLanglet/Twig-CS-Fixer/pull/218)

**Comment by VincentLanglet** [comment]
For context it was added by @smnandre in symfony/ux
https://github.com/symfony/ux/issues/1166
https://github.com/symfony/ux/pull/1793

**Comment by smnandre** [comment]
> For context it was added by @smnandre in symfony/ux

... only on the website (for now)  :) 

--- 

Regarding the suggestion, did you run it on the Symfony codebase ? Does it change a lot of files ? Only in "test / fixtures" or in Resources also ? 

That'd be nice to know the "impact" i think :)

**Comment by VincentLanglet** [comment]
> Regarding the suggestion, did you run it on the Symfony codebase ? Does it change a lot of files ? Only in "test / fixtures" or in Resources also ?
> 
> That'd be nice to know the "impact" i think :)

I didn't run it everywhere but I already found some issues like https://github.com/symfony/symfony/issues/54051

**Comment by derrabus** [comment]
> May I propose to add the tool `twig-cs-fixer` ([repo](https://github.com/VincentLanglet/Twig-CS-Fixer)) in this repo as a dev dependency

None of the tools we use (PHPUnit, PHP-CS-Fixer, Psalm) is an actual dev dependency on this repo. I don't think we want to change that.

**Comment by 94noni** [comment]
> actual dev dependency on this repo

yes I perhaps miswrite my purpose, more like adding a `twig-cs-fix` config file in the repo as its done for the `php-cs-fixer`
and from time to time, run it or via CI

> did you run it on the Symfony codebase ?

nope, but I can of course and report back here when I 've done it

**Comment by 94noni** [comment]
with a default config file, and the latest version of the fixer
on symfony/symfony 7.1, only src/Bundles
https://github.com/94noni/symfony/commit/5f57bab21568f34f550e0ced1b2ced9dbb6b2d15



looks like major things are spacing inside arrays and around operators:

```diff
- {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', { link: profiler_url }) }}
+ {{ include('@WebProfiler/Profiler/toolbar_item.html.twig', {link: profiler_url}) }}
```

and 

```diff
- {{ profiler_dump(usage.trace, maxDepth=2) }}
+ {{ profiler_dump(usage.trace, maxDepth = 2) }}
```


wdyt?

**Comment by chalasr** [comment]
Should it be [fabbot](https://fabbot.io)'s responsibility? 
<img width="600" alt="Screenshot 2024-05-11 at 17 09 27" src="https://github.com/symfony/symfony/assets/7502063/65eecc85-eab9-4b51-9b27-5bee6781f6bd">

**Comment by smnandre** [comment]
@chalasr If so, it should probably be Symfony specific then (at least at first) ?

**Comment by chalasr** [comment]
@smnandre Perhaps yes, I don't know much about fabbot internals as it's not public code yet. I'd expect it to run tools like php-cs-fixer only if a matching config file exists in the project. I may be wrong though.
What I'm pretty sure is, if we check twig files as part of the CI, only the files that changed in the given PR/commit should be checked (which is how fabbot operates).

**Comment by 94noni** [comment]
lets close for now, perhaps will be relevant when/if twig/twig comes inside this repo as symfony/twig later on
thx anyway for commenting here

**Comment by 94noni** [comment]
Friendly ping to commentors :)

With recent twig all new features etc, i really think it could be relevant to add such tool in main repo

Wdyt of restarting this ?
Previous diff run on 7.1 https://github.com/94noni/symfony/commit/5f57bab21568f34f550e0ced1b2ced9dbb6b2d15

**Comment by 94noni** [comment]
PR landed at https://github.com/symfony/symfony/pull/60761
